### ISSUE TITLE
Ancient INI_LoadFile bug

### ISSUE THREAD

**Issue by tr1xy** [pawn-lang/YSI-Includes#499]
Hello, just to clarify that I am no longer interested in SA-MP nor anything related to it (well yeah, except the burgershot.gg forum).

I've noticed [this](https://www.burgershot.gg/showthread.php?tid=1493) bug a while ago (probably years) but I was too lazy to report it and now that I've noticed that other users are having this issue - I'm reporting it. The bug is probably persistent in ancient old versions of YSI and again - I'm sorry for not reporting it before.

TL;DR version of the thread: You can't use anything except for the INI load functions in the callback that's being called by the INI_LoadFile. (you CAN but the file won't be loaded properly (at all))

**Comment by Y-Less** [comment]
That's not actually a bug, it is a design decision.  A very bad, unintuitive, decision, but one non-the-less.  The loading callback is called once for every key/value pair in the INI file, in an undefined order; thus there is no way to know when loading one value if all the other required values have already been loaded.  `INI_Int` etc are actually macros that look something like:

```pawn
#define INI_Int(%0,%1) \
	if (!strcmp(name, #%0)) {           \
		%1 = strval(value);             \
		return 1;                       \
	}
```

Why is this bad?  Oh so many reasons!

1. It is a macro that doesn't look like a macro, and in no way indicates to the user that it does anything other than a normal function call.
2. It has an embedded `return` within it, which means it might actually end the current function (hence why the code after isn't called).
3. It assumes there are variables called `name` and `value` available.  Always, without question.

So, yeah, possibly some of the worst code in YSI.  The original idea was you use `INI_Load` to load the data, THEN you use it - I didn't intend the callbacks to use the data being loaded (because, as I said, it is called once for every entry so if you need, say, five different entries to create a house you can't get them all in one call).  Having said that, there are still WAY better ways to do this (as seen in y_xml and later the various SQL plugins).
